tests/foreman_compute_resources_test.py

```diff
@@ -0,0 +1,10 @@
+import pytest
+
+FOREMAN_HOST = 'localhost'
+FOREMAN_PORT = 3000
+
+@pytest.mark.parametrize("compute_resource", ['AzureRm', 'EC2', 'GCE', 'Libvirt', 'Openstack', 'Vmware'])
+def test_foreman_compute_resources(server, compute_resource):
+    hammer = server.run("hammer compute-resource create --help | grep provider")
+    assert hammer.succeeded
+    assert compute_resource in hammer.stdout
```

Comment:
I was about to suggest to use the API directly (thus saving the requirement to have hammer set up), but the code gets ugly:

```
>>> next(pp for pp in next(p for p in foremanapi.resource('compute_resources').action('create').params if p.name == 'compute_resource').params if pp.name == 'provider').description
'\nProviders include Libvirt, Ovirt, EC2, Vmware, Openstack, AzureRm, GCE\n'
```

Reply:
For the non-core ones you could call `foremanapi.list('plugins')`, something like:
```python
plugins = [plugin['name'] for plugin in foremanapi.list('plugins')]
assert 'foreman_google' in plugins
```

Reply:
added [tests/foreman_plugins.py](https://github.com/theforeman/foremanctl/pull/344/commits/3ac369e1a7f833270618edbc7ab1970049edcdbc)